packages/auth/src/grant/model.ts

```diff
@@ -61,6 +66,70 @@ export class Grant extends BaseModel {
   public interactNonce?: string // AS-generated nonce for post-interaction hash
 }
 
+interface ToOpenPaymentsInteractiveGrantArgs {
+  authServerUrl: string
+  client: {
+    name: string
+    uri: string
+  }
+  waitTimeSeconds?: number
+}
+
+export function toOpenPaymentsInteractiveGrant(
+  grant: Grant,
+  args: ToOpenPaymentsInteractiveGrantArgs
+): OpenPaymentsInteractiveGrant {
+  if (!isInteractiveGrant(grant)) {
+    throw new Error('Expected interactive grant')
+  }
+
+  const { authServerUrl, client, waitTimeSeconds } = args
+
+  const redirectUri = new URL(
+    authServerUrl + `/interact/${grant.interactId}/${grant.interactNonce}`
+  )
+
+  redirectUri.searchParams.set('clientName', client.name)
+  redirectUri.searchParams.set('clientUri', client.uri)
+
+  return {
+    interact: {
+      redirect: redirectUri.toString(),
+      finish: grant.interactNonce
+    },
+    continue: {
+      access_token: {
+        value: grant.continueToken
+      },
+      uri: `${authServerUrl}/auth/continue/${grant.continueId}`,
+      wait: waitTimeSeconds
+    }
+  }
+}
+
+interface ToOpenPaymentsGrantArgs {
+  authServerUrl: string
+}
+
+export function toOpenPaymentsNonInteractiveGrant(
```

Comment:
I'm not sure I'm fully convinced of having `NonInteractiveGrant` and `InteractiveGrant`.
Maybe we could just have `Grant` and `InteractiveGrant`, it still could be clear but less wordy...thoughts?

Reply:
We could be inconsistent with everywhere else in rafiki and have `Grant` and `PendingGrant`
https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol#section-1.5

Reply:
I wouldn't mind updating to `Grant` and `PendingGrant` everywhere